pkg/utils/command.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import "strings"
+
+// MergeCommands merges injected commands and original commands for injecting commands to containers,
+// eg: inject: []string{"bash", "-c", "/check.sh"}, origin: []string{"bash", "-c", "/run.sh"}
+// merged commands: []string{"/bin/sh", "-ec", "/check.sh\n/run.sh"}
+func MergeCommands(inject []string, origin []string, args []string) []string {
+	// merge injected commands
+	scripts := mergeCommandsAction(inject)
+
+	// merge original commands
+	scripts += mergeOriginCommandsAndArgs(origin, args)
+
+	return []string{"/bin/sh", "-ec", scripts}
+}
+
+func mergeOriginCommandsAndArgs(origin []string, args []string) string {
+	commands := []string{}
+	commands = append(commands, origin...)
+	commands = append(commands, args...)
+
+	return mergeCommandsAction(commands)
+}
+
+func mergeCommandsAction(commands []string) string {
```

Comment:
Can we split this function, it too long

Reply:
This is a relatively complete logic. If it is broken into small functions, the function name is not good.